enforcer-rules/src/main/java/io/jenkins/tools/incrementals/enforcer/RequireExtensionVersion.java

```diff
@@ -25,32 +25,60 @@
 package io.jenkins.tools.incrementals.enforcer;
 
 import java.util.List;
+import java.util.Objects;
 import java.util.regex.Matcher;
 import java.util.regex.Pattern;
+import javax.inject.Inject;
+import javax.inject.Named;
 import org.apache.maven.AbstractMavenLifecycleParticipant;
+import org.apache.maven.artifact.versioning.ArtifactVersion;
 import org.apache.maven.artifact.versioning.DefaultArtifactVersion;
+import org.apache.maven.artifact.versioning.InvalidVersionSpecificationException;
+import org.apache.maven.artifact.versioning.VersionRange;
+import org.apache.maven.enforcer.rule.api.AbstractEnforcerRule;
 import org.apache.maven.enforcer.rule.api.EnforcerRuleException;
-import org.apache.maven.enforcer.rule.api.EnforcerRuleHelper;
-import org.apache.maven.plugin.logging.Log;
-import org.apache.maven.plugins.enforcer.AbstractVersionEnforcer;
+import org.codehaus.plexus.PlexusContainer;
 import org.codehaus.plexus.classworlds.realm.ClassRealm;
 import org.codehaus.plexus.component.repository.exception.ComponentLookupException;
+import org.codehaus.plexus.util.StringUtils;
 
 /**
  * Verifies that {@code git-changelist-maven-extension}, if present, is sufficiently new.
  */
-public class RequireExtensionVersion extends AbstractVersionEnforcer {
+@Named("requireExtensionVersion")
```

Comment:
As recommended upstream. This means we'll need to change `<rule implementation="io.jenkins.tools.incrementals.enforcer.RequireExtensionVersion">` to `<requireExtensionVersion>` in consumers, which looks nicer anyway.